launcher/InstanceImportTask.cpp

```diff
@@ -425,8 +433,52 @@ void InstanceImportTask::processFlame()
         m_modIdResolver.reset();
         connect(m_filesNetJob.get(), &NetJob::succeeded, this, [&]()
         {
+            qDebug() << "Finished with blocked mods : " << m_blockedMods.size();
             m_filesNetJob.reset();
-            emitSucceeded();
+            if(!m_blockedMods.empty())
+            {
+                //blocked mods found, we need the slug for displaying.... we need another job :D !
+                auto slugJob = new NetJob("Slug Job", APPLICATION->network());
+                auto slugs = QVector<QByteArray>(m_blockedMods.size());
+                auto index = 0;
+                for(const auto& result: m_blockedMods)
+                {
+                    auto projectId = result.projectId;
+                    slugs[index] = QByteArray();
+                    auto url = QString("https://api.curseforge.com/v1/mods/%1").arg(projectId);
+                    auto dl = Net::Download::makeByteArray(url, &slugs[index]);
+                    slugJob->addNetAction(dl);
+                    index++;
+                }
+                connect(slugJob, &NetJob::succeeded, this, [slugs,this,slugJob]()
+                {
+                    slugJob->deleteLater();
+                    QString text;
+                    auto index = 0;
+                    for(const auto& result: slugs)
+                    {
+                        auto json = QJsonDocument::fromJson(result);
+                        auto slug = json.object().value("data").toObject().value("slug").toString();
+                        auto mod = m_blockedMods[index];
+                        auto link = QString("https://www.curseforge.com/minecraft/mc-mods/%1/download/%2").arg(slug,QString::number(mod.fileId));
+                        text += QString("%1: <a href='%2'>%2</a><br/>").arg(mod.fileName,link);
+                        index++;
+                    }
+                    qWarning() << "Blocked mods found, displaying mod list";
+                    auto message = CustomMessageBox::selectable(nullptr, tr("Blocked mods found"),
+                                                                tr("The following mods were blocked on third party launchers<br/>"
+                                                                   "You will need to manually download them and add them to the modpack")
+                                                                + "<br/><br/>" + text,
+                                                                QMessageBox::Warning, QMessageBox::Ok, QMessageBox::Ok);
+                    message->setTextFormat(Qt::RichText);
+                    message->exec();
+                    m_blockedMods.clear();
+                    emitSucceeded();
+                });
+                slugJob->start();
+            }else {
+                emitSucceeded();
+            }
```

Comment:
You could move this to `FileResolvingTask` for fetching this info for *all* mods. Might be useful later.